hack/generated/pkg/reconcilers/deployment_error_translator.go

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package reconcilers
+
+import (
+	"github.com/Azure/go-autorest/autorest/azure"
+
+	"github.com/Azure/azure-service-operator/hack/generated/pkg/armclient"
+)
+
+func stringOrEmptyString(value interface{}) string {
```

Comment:
Does this need to support `fmt.Stringer` too?

Reply:
If this were going to be an exported method or used in other contexts then yea, supporting `fmt.Stringer` seems like it'd be a good idea.

In this context my intent was really as a helper method for parsing the horrible `[]map[string]interface{}` error from the Azure SDK and so it has a pretty narrow expected usage context. If it was going to be used more broadly I think we'd also want it to not silently allow non-string (and non `fmt.Stringer`) and just turn those into `""`. That case feels more like an `err` possibly? Adding an `err` though muddies the actual usage as then I'd have to discard it (see where it's called down below for what I mean).

Reply:
I've added a comment to this method explaining its behavior (which I'm hoping will help steer anybody else in this package away from it unless they want this specific behavior)

Reply:
Looks good.